web-components/src/components/form/BatchBasicsForm.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { useState } from 'react';
+import { Formik, Form, Field, FieldArray } from 'formik';
+import { Box, IconButton, Link } from '@mui/material';
+import { useTheme, DefaultTheme as Theme } from '@mui/styles';
+import { string, date, object, AnyObjectSchema } from 'yup';
+
+import { Body } from '../typography';
+import InputLabel from '../inputs/InputLabel';
+import OnBoardingCard from '../cards/OnBoardingCard';
+import TextField from '../inputs/TextField';
+import ControlledTextField from '../inputs/ControlledTextField';
+import SelectTextField from '../inputs/SelectTextField';
+import { DatePickField } from '../inputs/DatePickField';
+import SaveFooter from '../fixed-footer/SaveFooter';
+import OutlinedButton from '../buttons/OutlinedButton';
+import TrashIcon from '../icons/TrashIcon';
+import { AddCertificationModal } from '../modal/AddCertificationModal';
+import {
+  requiredMessage,
+  invalidDate,
+  vcsRetirementSerialRE,
+  invalidVCSRetirement,
+  invalidVCSID,
+  numericOnlyRE,
+} from '../inputs/validation';
+import { NameUrl } from '../../types/rdf';
+import { VCSBatchMetadataLD } from '../../types/rdf/C01-verified-carbon-standard-batch';
+
+export interface BatchBasicsFormValues {
+  class_id: string;
+  start_date: Date | null;
+  end_date: Date | null;
+  metadata?: Partial<VCSBatchMetadataLD>;
+  metadata_json?: string;
```

Comment:
why are we using snake case by the way?

Reply:
matching to what would go in a ledger message. `metadata_json` doesn't really make sense, but I figured it would be reassigned to `metadata`.

Reply:
the ledger API returns snake case formatted responses but we can't use that for submitting transactions, see https://github.com/regen-network/regen-js/blob/main/packages/api/src/generated/regen/ecocredit/v1alpha1/tx.ts#L30